plugins/ec2/crossVpcPublicPrivate.js

```diff
@@ -0,0 +1,222 @@
+                            });
+                        }
+                    }
+                }
+                for (i in subVpc) {
+                    for (j in tempPubSubnets) {
+                        if (subVpc[i].subId == tempPubSubnets[j]) {
+                            subnets.push({
+                                subId: subVpc[i].subId,
+                                subCidr: subVpc[i].subCidr,
+                                subVpcId: subVpc[i].vcpId,
+                                type: 'public'
+                            });
+                        }
+                    }
+                }
+
+                // filtering the routes that are related to the peering process
+                var peeringRoutes = [];
+
+                for (i in vpcPeeringInfo) {
+                    for (j in routesInfo) {
+                        if (vpcPeeringInfo[i].peeringId == routesInfo[j].rtVpcId) {
+                            peeringRoutes.push({
+                                routeId: routesInfo[j].rtId,
+                                vpcId: vpcPeeringInfo[i].vpcId,
+                                peeringCidr: routesInfo[j].rtCB,
+                                peeringId: vpcPeeringInfo[i].peeringId
+                            });
+                        }
+                    }
+                }
+
+                // generating the public and private records
+                var pubRecord = [];
+                var prvRecord = [];
+
+                for (i in peeringRoutes) {
+                    for (j in subnets) {
+                        if (peeringRoutes[i].vpcId == subnets[j].subVpcId && peeringRoutes[i].peeringCidr == subnets[j].subCidr) {
+                            if (subnets[j].type == 'public') {
+                                pubRecord.push({
+                                    vpcId: peeringRoutes[i].vpcId,
+                                    peeringId: peeringRoutes[i].peeringId,
+                                    subnetId: subnets[j].subId
+                                });
+                            } else {
+                                prvRecord.push({
+                                    vpcId: peeringRoutes[i].vpcId,
+                                    peeringId: peeringRoutes[i].peeringId,
+                                    subnetId: subnets[j].subId
+                                });
+                            }
+                        }
+                    }
+                }
+
+                // comparing and showing the results
+                for (i in pubRecord) {
+                    for (j in prvRecord) {
+                        if (pubRecord[i].peeringId == prvRecord[j].peeringId) {
+                            helpers.addResult(results, 2, 'Public-Private cross vpc subnet connection found, for Subnets: ' + pubRecord[i].subnetId + ' and ' + prvRecord[j].subnetId, region);
+                        }
+                    }
+                }
```

Comment:
Keep track of results above and if there are none for this region, at the end, add a passing result:
```
helpers.addResult(results, 0, 'No routes between public and private subnets of different VPCs found', region);
```

Reply:
this is done